app/controllers/users_controller.rb

```diff
@@ -18,6 +18,10 @@ def destroy
     redirect_to users_path
   end
 
+  def admin_panel
+    authorize User
```

Comment:
How about doing it in a separate controller instead of a custom action? It's more REST-friendly